core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -10,14 +12,48 @@
 logger = logging.getLogger("dy")
 
 
+@unique
+class OpCode(Enum):
+    CREATE = "c"
+    DELETE = "d"
+    UPDATE = "u"
+
+
+def get_fields(mapping: Optional[Union[Dict[str, Any], str]]) -> List[Dict[str, Any]]:
+    return [{"column": str(next(iter(item.keys()))),
+             "key": str(next(iter(item.values())))}
+            if isinstance(item, dict) else {"column": item, "key": item} for item in mapping] if mapping else []
+
+
+def generate_upsert_stmt(
+        table: str, primary_keys: List[Dict[str, Any]],
+        mapping_fields: List[Dict[str, Any]],
+        db_type: str) -> Any:
+    if db_type.lower() == "postgresql":
+        update_fields = [f"{field['column']} = {sa.bindparam(field['key'])}" for field in mapping_fields]
+
+        insert_fields = ", ".join([field["column"] for field in mapping_fields])
+        pk_fields = ", ".join([field["column"] for field in primary_keys])
+
+        insert_bind_params = ", ".join([f"{sa.bindparam(field['key'])}" for field in mapping_fields])
+
+        return sa.text(f"""INSERT INTO {table} ({insert_fields})
```

Comment:
there is an sql builder for the upsert now in SA. https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#insert-on-conflict-upsert.
we should move to the API in next iteration as we add more dialects.

Reply:
I'll open an issue for that.